src/sysfiles.h

```diff
@@ -339,7 +339,7 @@ UInt SyExecuteProcess(Char * dir, Char * prg, Int in, Int out, Char * args[]);
 **
 *F  SyIsExistingFile( <name> )  . . . . . . . . . . . does file <name> exists
 **
-**  'SyIsExistingFile' returns 1 if the  file <name> exists and 0  otherwise.
+**  'SyIsExistingFile' returns 0 if the  file <name> exists and -1  otherwise.
```

Comment:
Arrgh! OK, first off: good catch.

Secondly: This is a case were I very much think the documentation is right and code is wrong. Reading code that does `if (SyIsExistingFile(name)) { ... do stuff that assumes the files does NOT exist ... }` is just begging for somebody to get confused and screw it up.

So, I'd prefer if you didn't "fix" the comments here, and instead we fix those function (and all code calling them) to have sane return values which match the name of the function. In fact, I think we should use `BOOL`, `TRUE` and `FALSE` for them.

(To be clear, I am not asking you to make that fix which goes beyond what you wanted to do in this PR)

Reply:
Looking closer, it's actually worse, as the return value of `SyIsExistingFile` actually does not allow distinguishing between "the file does not exist" and "some error occurred".

Honestly, I wonder why we even bother with these "abstraction layers". They don't have well-defined semantics, and each typically has only a single caller (`SyIsReadableFile` being the exception). So a hypothetical porting effort wouldn't really benefit from their existence... I'll think about submitting a PR to just merge them into their callers.